item-impl/src/main/java/com/example/auction/item/impl/ItemRepository.java

```diff
@@ -0,0 +1,248 @@
+package com.example.auction.item.impl;
+
+import akka.Done;
+import com.datastax.driver.core.*;
+import com.datastax.driver.extras.codecs.enums.EnumNameCodec;
+import com.example.auction.item.api.ItemStatus;
+import com.example.auction.item.api.ItemSummary;
+import com.example.auction.item.api.PaginatedSequence;
+import com.lightbend.lagom.javadsl.persistence.AggregateEventTag;
+import com.lightbend.lagom.javadsl.persistence.ReadSide;
+import com.lightbend.lagom.javadsl.persistence.ReadSideProcessor;
+import com.lightbend.lagom.javadsl.persistence.cassandra.CassandraReadSide;
+import com.lightbend.lagom.javadsl.persistence.cassandra.CassandraSession;
+import org.pcollections.PSequence;
+import org.pcollections.TreePVector;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.stream.Collectors;
+
+import static com.example.auction.item.impl.CompletionStageUtils.*;
+
+@Singleton
+public class ItemRepository {
+
+    private final CassandraSession session;
+
+    @Inject
+    public ItemRepository(CassandraSession session, ReadSide readSide) {
+        this.session = session;
+        session.underlying().thenAccept(s -> registerCodec(s, new EnumNameCodec<>(ItemStatus.class)));
```

Comment:
Is this a good place to register codecs for the Cassandra Java driver? It might be nice to offer a convenience for that.